src/core/system_titles.cpp

```diff
@@ -0,0 +1,1124 @@
+                                               0x00000003, 0x00000003, 0x00000003},
+                         },
+                         {
+                             .name = "SAFE_MODE_FIRM (N3DS)",
+                             .sets = SystemTitleSet::New3ds | SystemTitleSet::Minimal,
+                             .title_id_lows = {0x20000003, 0x20000003, 0x20000003, 0x20000003,
+                                               0x20000003, 0x20000003, 0x20000003},
+                         },
+                         {
+                             .name = "TWL_FIRM (O3DS)",
+                             .sets = SystemTitleSet::Old3ds,
+                             .title_id_lows = {0x00000102, 0x00000102, 0x00000102, 0x00000102,
+                                               0x00000102, 0x00000102, 0x00000102},
+                         },
+                         {
+                             .name = "TWL_FIRM (N3DS)",
+                             .sets = SystemTitleSet::New3ds,
+                             .title_id_lows = {0x20000102, 0x20000102, 0x20000102, 0x20000102,
+                                               0x20000102, 0x20000102, 0x20000102},
+                         },
+                         {
+                             .name = "AGB_FIRM (O3DS)",
+                             .sets = SystemTitleSet::Old3ds,
+                             .title_id_lows = {0x00000202, 0x00000202, 0x00000202, 0x00000202,
+                                               0x00000202, 0x00000202, 0x00000202},
+                         },
+                         {
+                             .name = "AGB_FIRM (N3DS)",
+                             .sets = SystemTitleSet::New3ds,
+                             .title_id_lows = {0x20000202, 0x20000202, 0x20000202, 0x20000202,
+                                               0x20000202, 0x20000202, 0x20000202},
+                         },
+                     }},
+            }};
+
+std::vector<u64> GetSystemTitleIds(SystemTitleSet set, u32 region) {
+    std::vector<u64> title_ids;
+    for (const auto& category : system_titles) {
+        for (const auto& title : category.titles) {
+            if ((title.sets & set) != 0 && title.title_id_lows[region] != 0) {
+                title_ids.push_back((static_cast<u64>(category.title_id_high) << 32) |
+                                    static_cast<u64>(title.title_id_lows[region]));
+            }
+        }
+    }
+
+    return title_ids;
+}
+
+u64 GetHomeMenuTitleId(u32 region) {
+    return (static_cast<u64>(home_menu_title_id_high) << 32) |
+           static_cast<u64>(home_menu_title.title_id_lows[region]);
+}
+
+std::string GetHomeMenuNcchPath(u32 region) {
+    return Service::AM::GetTitleContentPath(Service::FS::MediaType::NAND,
+                                            GetHomeMenuTitleId(region));
+}
+
+/// Returns the region of a system title, or -1 if it could not be determined.
+int GetSystemTitleRegion(u64 title_id) {
```

Comment:
```suggestion
std::optional<int> GetSystemTitleRegion(u64 title_id) {
```

rather than returning -1 for an invalid match